lib/services/authority.go

```diff
@@ -210,24 +210,27 @@ func checkSAMLIDPCA(cai types.CertAuthority) error {
 		return trace.BadParameter("unknown CA type %T", cai)
 	}
 
-	if len(ca.Spec.ActiveKeys.TLS) != 1 {
-		return trace.BadParameter("SAML IDP certificate authority must contain one key pair")
+	if len(ca.Spec.ActiveKeys.TLS) == 0 {
+		return trace.BadParameter("missing SAML IdP CA")
 	}
 
-	pair := ca.GetTrustedTLSKeyPairs()[0]
-	if len(pair.Key) != 0 && pair.KeyType == types.PrivateKeyType_RAW {
-		var err error
-		if len(pair.Cert) > 0 {
-			_, err = tls.X509KeyPair(pair.Cert, pair.Key)
-		} else {
-			_, err = utils.ParsePrivateKey(pair.Key)
+	for _, pair := range ca.GetTrustedTLSKeyPairs() {
+		if len(pair.Key) != 0 && pair.KeyType == types.PrivateKeyType_RAW {
+			var err error
+			if len(pair.Cert) > 0 {
+				_, err = tls.X509KeyPair(pair.Cert, pair.Key)
+			} else {
+				_, err = utils.ParsePrivateKey(pair.Key)
+			}
+			if err != nil {
+				return trace.Wrap(err)
+			}
 		}
-		if err != nil {
+		if _, err := tlsca.ParseCertificatePEM(pair.Cert); err != nil {
```

Comment:
Isn't this redundant if we do `tls.X509KeyPair` above?

Reply:
tls.X509Keypair does some additional checks that may not apply in the situation that the key is not checkable. However, it would probably make sense to put this in an else so that this only happens in the situation where tls.X509KeyPair isn't executed.